### PR TITLE
Fix json serialization error if meilisearch response contains facet_stats

### LINKED ISSUE #474: Error parsing response JSON for facetStats
**Description**
If response from Mailisearch contains facetStats (facet field must contain numeric values to make it  happen), sdk returns error. It seems it happens because FacetStats struct is defined with whole number field types. But meilisearch response json contains floats. Changing u32 to f64 fixes this problem.
```
pub struct FacetStats {
    pub min: u32,
    pub max: u32,
}
```

**Expected behavior**
No error should happen.

**Current behavior**
Error parsing response JSON returned

**Environment (please complete the following information):**
- OS: Linux
- Meilisearch version: 1.1
- meilisearch-rust version: 0.23.1

### PR DESCRIPTION
**Pull request by Vedmak**
# Pull Request

## Related issue
Fixes #474

## What does this PR do?
- change field types of FacetStats struct from u32 to f64
- add test for facet_stats

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

### CODE REVIEW

File: src/search.rs [hunk 1]
```diff
@@ -50,8 +50,8 @@ pub struct SearchResult<T> {
 #[derive(Deserialize, Debug, Clone)]
 #[serde(rename_all = "camelCase")]
 pub struct FacetStats {
-    pub min: u32,
-    pub max: u32,
+    pub min: f64,
+    pub max: f64,
```
Comment (bidoubiwa) [inline src/search.rs:54]: does `u64` not fix this issue? I don't think facet stats can return a negative number

**Review by bidoubiwa — COMMENTED** [review]
Thanks a lot for this contribution 🙏

File: src/search.rs (hunk 1 above)
Comment (sanders41) [reply src/search.rs:54]: `u64` is an unsigned integer so according to issue it would still error with this since it needs a decimal.

Comment (bidoubiwa) [reply src/search.rs:54]: Oh yeah ... my bad 🙈

Comment (Vedmak) [reply src/search.rs:54]: Looked in meilisearch repo, and there this struct has f64 fields https://github.com/search?q=repo%3Ameilisearch%2Fmeilisearch%20FacetStats&type=code

**Review by bidoubiwa — APPROVED** [review]

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**